lib/Driver/ToolChain.cpp

```diff
@@ -62,6 +62,27 @@ ToolChain::JobContext::getTemporaryFilePath(const llvm::Twine &name,
   return C.getArgs().MakeArgString(buffer.str());
 }
 
+ToolChain::ResponseFileInfo ToolChain::getResponseFileInfoIfNeeded(
```

Comment:
Good call. I originally did that, but it didn't seem worth it because I didn't also change the `Job` constructor signature to use it, so the call sites actually became more complicated. After passing the optional throughout `Job`/`BatchJob` as well, it came out much cleaner.

Reply:
Cool!